tictactoe.rb

```diff
@@ -1,184 +1,185 @@
+		  	return false
+		  elsif @board[row][column] != "-"
+				puts "You cannot overwrite another move. Try Again you fucking idiot."
+				return false
+		  else
+		  	return true
+		  end
+		end
 
-	def winner
-		i = 0
-		winner_logic = false
-		while i < @board.length
-			if (@board[i][0] == @board[i][1]) && (@board[i][1] == @board[i][2]) && (@board[i][0] != "-")
-				# puts "#{@board[i][0]} is the winner!!!"
-				winner_logic = true
+		def winner
+			i = 0
+			winner_logic = false
+			while i < @board.length
+				if (@board[i][0] == @board[i][1]) && (@board[i][1] == @board[i][2]) && (@board[i][0] != "-")
+					# puts "#{@board[i][0]} is the winner!!!"
+					winner_logic = true
+				end
+				i += 1
 			end
-			i += 1
-		end
-		j = 0
-		while j < @board.length
-			if @board[0][j] == @board[1][j] && @board[1][j] == @board[2][j] && @board[0][j] != "-"
-				# puts "#{@board[0][j]} is the winner!!!"
+			j = 0
+			while j < @board.length
+				if @board[0][j] == @board[1][j] && @board[1][j] == @board[2][j] && @board[0][j] != "-"
+					# puts "#{@board[0][j]} is the winner!!!"
+					winner_logic = true
+				end
+				j += 1
+			end
+			if @board[0][0] == @board[1][1] && @board[1][1] == @board[2][2] && @board[0][0] != "-"
+				# puts "#{@board[0][0]} is the winner!!!"
+				winner_logic = true
+			elsif @board[2][0] == @board[1][1] && @board[1][1] == @board[0][2] && @board[2][0] != "-"
+				# puts "#{@board[2][0]} is the winner!!!"
 				winner_logic = true
 			end
-			j += 1
+			return winner_logic
 		end
-		if @board[0][0] == @board[1][1] && @board[1][1] == @board[2][2] && @board[0][0] != "-"
-			# puts "#{@board[0][0]} is the winner!!!"
-			winner_logic = true
-		elsif @board[2][0] == @board[1][1] && @board[1][1] == @board[0][2] && @board[2][0] != "-"
-			# puts "#{@board[2][0]} is the winner!!!"
-			winner_logic = true
-		end
-		return winner_logic
-	end
 
 
-	def board_not_full
-		logic = @board.any? do |x|
-			x.include?("-")
+		def board_not_full
+			logic = @board.any? do |x|
+				x.include?("-")
 
+			end
 		end
-	end
 
-	def stale_mate
-		puts "No moves left"
-		puts "D R A W"
+		def stale_mate
+			puts "No moves left"
+			puts "D R A W"
+		end
 	end
-end
 
-new_board = Board.new 
+	new_board = Board.new 
+
+	new_game = Game.new()
+	new_game.start_game
 
-new_game = Game.new()
-new_game.start_game
+	while !new_board.winner && new_board.board_not_full
+		new_board.display
+		new_game.turn
+		new_board.make_move(new_game.player_turn)
+		new_game.switch_turn
+	end
 
-while !new_board.winner && new_board.board_not_full
 	new_board.display
-	new_game.turn
-	new_board.make_move(new_game.player_turn)
-	new_game.switch_turn
-end
-
-new_board.display
-
-# if new_board.winner
-# 	new_board.winner
-# elsif !new_board.board_not_full
-# 	new_board.stale_mate
-# end
-
-# start game
-# set new board
-# random player goes first
-# make move       <----
-# check for win        |
-# next player      ----
+
+	# if new_board.winner
```

Comment:
You could have the run instructions in their own file 'run.rb' and run that file from the command line